controllers/node_kernel_reconciler.go

```diff
@@ -37,7 +38,14 @@ func (r *NodeKernelReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		return ctrl.Result{}, fmt.Errorf("could not get node: %v", err)
 	}
 
+	// We expect an osVersion of the form 411.86.202210072320-0 for example
+	osVersionRegexp := regexp.MustCompile("[0-9]+.[0-9]+.[0-9]+-[0-9]")
```

Comment:
`.` will match any character, not just dots.
```suggestion
	osVersionRegexp := regexp.MustCompile(`\d+\.\d+\.\d+\-\d`)
```
We should also move this outside of the function so it's ran only once.